parser/Parser.cc

```diff
@@ -24,14 +24,29 @@ class ErrorToError {
         return min((uint32_t)(pos), maxOff);
     }
 
-    static void maybeAddAutocorrect(core::ErrorBuilder &e, core::Loc loc, ruby_parser::dclass errorClass) {
+    static void maybeAddAutocorrect(core::GlobalState &gs, core::ErrorBuilder &e, core::Loc loc,
+                                    ruby_parser::dclass errorClass) {
         switch (errorClass) {
             case ruby_parser::dclass::IfInsteadOfItForTest:
                 e.replaceWith("Replace with `it`", loc, "it");
                 break;
             case ruby_parser::dclass::MissingCommaBetweenKwargs:
                 e.replaceWith("Insert a comma", loc, ", ");
                 break;
+            case ruby_parser::dclass::CurlyBracesAroundBlockPass: {
+                auto source = loc.source(gs);
+                if (source.has_value()) {
```

Comment:
Is this check necessary? Won't the source always exist as the loc comes from the file being parsed?

Reply:
True, I imagine it's not strictly required. What do you think about switching this to an `ENFORCE`?

Reply:
No, this is very much intentional

Reply:
I made `.source()` return `nullopt` because it was a common source of crashes

Reply:
#4075

Reply:
ah i see what you're saying, we know exactly where the `loc` comes from in this particular case.

Even still, i'd kind of prefer writing it this way because i don't want people to cargo-cult that you should always `ENFORCE` when using `.source`, if you don't mind.

Reply:
Works for me!

Reply:
Works for me!